lib/AnalysisStructured/PtrAnalysis.cpp

```diff
@@ -691,6 +737,65 @@ LogicalResult PtrAnalysis::rewriteAddptrOp(triton::AddPtrOp op) {
   return success();
 }
 
+LogicalResult PtrAnalysis::rewriteMakeTensorPtrOp(triton::MakeTensorPtrOp op) {
+  OpBuilder builder(op);
+
+  PtrState state;
+  if (visitOperandMakeTensorPtr(op, state, op.getLoc(), builder).failed()) {
+    return failure();
+  }
+
+  auto maketptrOp = state.createTTSMakeTensorPtrOp(builder, op.getLoc());
+  knownPtrs[op.getResult()] = state;
+  ptrMap.map(op.getResult(), maketptrOp.getResult());
+  return success();
+}
+
+LogicalResult PtrAnalysis::rewriteAdvanceOp(triton::AdvanceOp op) {
+  OpBuilder builder(op);
+  auto loc = op.getLoc();
+
+  PtrState state;
+  if (visitOperand(op->getOperand(0), state, loc, builder).failed()) {
+    op->emitRemark("PtrAnalysis: Failed to analyze ptr of tt.advance");
+    return failure();
+  }
+  assert(!state.order.empty() && "block pointer should have order field set");
```

Comment:
nit: `state.order.empty` can probably be refactored to a helper like `state.isBlockPtr()` or something to make things easier to read